src/pluginplay/printing/mermaid.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <pluginplay/module/module_class.hpp>
+#include <pluginplay/module_manager/module_manager_class.hpp>
+#include <sstream>
+
+namespace pluginplay::printing {
+
+void add_letter(std::string& code) {
+    for(int i = (code.size() - 1); i >= 0; --i) {
```

Comment:
```suggestion
    if(code.size() == 0) return;
    for(int i = (code.size() - 1); i >= 0; --i) {
```
You need to be very careful when subtracting from unsigned integers. If `code.size() == 0` this would try to set `i` to $2^{64}-1$ and this would run forever (though because you then cast it to a signed integer I think it will actually be undefined behavior because `int` can't hold $2^{64}-1$). It is always better to count up.